src/codegen/expression.rs

```diff
@@ -2667,7 +2667,7 @@ pub fn assign_single(
 
             Expression::Variable {
                 loc: left.loc(),
-                ty: ty.clone(),
+                ty: ty.deref_storage().clone(), // LHS value propagation is always in memory
                 var_no: pos,
```

Comment:
Although this works there is a mismatch between how variable `pos` is used, and how it is created on line 2584.

`--emit cfg --no-constant-folding --no-cse`
```
# function C::C::function::f__uint64 public:true selector:70e8dbcb nonpayable:true
# params: uint64 x
# returns: uint64
block0: # entry
	ty:uint64 %x = (arg #0)
	ty:uint64 storage %temp.11 = %x
	store storage slot(uint256 1) ty:uint64 = %temp.11
	ty:uint64 storage %temp.12 = %temp.11
	store storage slot(uint256 0) ty:uint64 = %temp.12
	return %temp.12
```
See how `%temp.11` and `%temp.12` are used as `uint64` but the variable type is `uint64 storage`.

Reply:
Hmm yeah, I already tried to switch to the RHS type, but that lead to a bunch of codegen test failures. Maybe those are wrong too

Reply:
I noticed that too, I think the codegen tests will need fixing up